lib/src/facts/linux/networking_resolver.cc

```diff
@@ -142,6 +142,17 @@ namespace facter { namespace facts { namespace linux {
         auto parse_route_line = [&known_route_types](string& line, std::vector<route>& routes) {
             vector<boost::iterator_range<string::iterator>> parts;
             boost::split(parts, line, boost::is_space(), boost::token_compress_on);
+
+            // remove trailing "onlink" or "pervasive" flags
+            while (parts.size() > 0) {
+                std::string last_token(parts.back().begin(),
```

Comment:
Could still use `auto last_token = parts.back()` here. It's automatically a reference because that's what `back()` returns.